src/v2.rs

```diff
@@ -34,13 +34,19 @@ enum HeaderFrameKind {
 
 /// Version 2 of the Eszip format. This format supports streaming sources and
 /// source maps.
+///
+/// # Clone
+///
+/// This struct supports cloning, but when its data is complete. If the
+/// `EszipV2` returned from [`Self::parse`] is cloned before the future
+/// successfully resolves, the clone operation will panic.
 #[derive(Debug, Default)]
 pub struct EszipV2 {
   modules: Arc<Mutex<HashMap<String, EszipV2Module>>>,
```

Comment:
I feel like this could just be a shallow clone of `modules` without needing to panic if the import map were a separate mutable field on here, but I'm not familiar with this code.

Reply:
Or maybe alternatively if the import map were provided when creating an EszipV2 it wouldn't be necessary either. Then we could remove all mutable methods and store everything in an `Arc`. Actually, why not just store `EszipV2` within an arc in the code that wants to clone this?

Reply:
My use case for allowing cloning is to enable web workers with `deno compile`. In this case, the eszip's modules would be loaded by the time the main worker starts running, and it would not change afterwards. However, the modules will be read from all workers, and having a mutex guarding the actual module map would cause contention. Which is why I thought it was better to enable cloning.

The alternative, as I see it, is to maybe have a `ReadOnlyEszipV2`, and have the read-only methods shared in a common trait. Should I work on that instead?

Reply:
I think doing a deep clone here would use more memory. It might be good to profile first because I'm not sure what the best strategy is here. For example, maybe it's not a big deal to just put it in an `Arc<EszipV2>` and share that between all the workers. Maybe using an `RwLock` instead of `Mutex` here would be better. Maybe taking out all the modules from the mutex would be best and having a `ReadOnlyEszipV2` would be better. I think it would be good to try this out and see which one is best in practice. Thoughts?

Reply:
It doesn't seem like a clone which clones an interior-mutable `Arc` of modules would be correct, whether the interior mutability happens with `Mutex` or `RwLock`, since `EszipV2::add_import_map` modifies both `modules` and `ordered_modules`. This would mean that a clone's state could become invalid by mutating the original, or vice versa.

I'll look into `ReadOnlyEszipV2`.

Reply:
Yeah, that's why I was wondering if the import map could just be provided when creating an `EszipV2`. Then nothing would need to be mutable I think. The `add_import_map` function is a little strange. Looking at deploy's code it seems like a mutable method is not necessary.

Reply:
Mutability is still necessary to be able to do a streaming build of the module source code

Reply:
But that's within the `modules: Arc<Mutex<` that could be shared, right?